repository-data/webfiles/src/main/resources/site/freemarker/common/macro/sections/tableauSection.ftl

```diff
@@ -15,20 +15,51 @@
         <div id="${divId}"></div>
 
         <script type="text/javascript">
-            window.addEventListener('load', initViz);
+            var viz;
 
-            function initViz() {
-                var containerDiv = document.getElementById("${divId}");
+            function throttleViz(containerDiv) {
+                <#if section.throttlingLocation??>
+                    <#local tableauThrottleSize="uk.nhs.digital.freemarker.tableau.RemoteThrottleSizeFromUrl"?new() />
+                    <#local remote = tableauThrottleSize(section.throttlingLocation) />
+                    var attemptProbability = ${remote.size};
+                <#else>
+                    var attemptProbability = 0.5;
+                </#if>
+
+                if (viz === undefined) {
+                    if ((Math.floor(Math.random() * 101) / 100.0) <= attemptProbability) {
+                        cookies.setCookie('tableauWithoutThrottling', 'yes', 0.042 <#-- One hour -->);
+                        containerDiv.innerHTML = ''
```

Comment:
While there aren't any security considerations with your usage here, you might get some security flags on the usage of innerHTML. See: https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#Security_considerations and https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText

Replacing it with `containerDiv.innerText = '';` should functionally be the same.

Reply:
Great spot